src/main/java/bowling/domain/frame/Frame.java

```diff
@@ -30,14 +36,63 @@ public Frame next() {
     }
 
     public void roll(int numberOfDownPin) {
-        this.pins.down(numberOfDownPin);
+        this.frameDownPins.down(numberOfDownPin);
     }
 
     public boolean hasTurn() {
-        return this.pins.hasTurn();
+        return this.frameDownPins.hasTurn();
     }
 
     public FrameResult getFrameResult() {
-        return new FrameResult(this.pins.getDownPins(), this.pins.getScoreType());
+        return new FrameResult(this.frameDownPins.getDownPins(), this.frameDownPins.getScoreType(), frameScore());
+    }
+
+    public Score frameScore() {
+        if (!isFinished()) {
+            return Score.create(0, ScoreType.READY);
+        }
+
+        if (isLast() && isFinished()) {
+            return Score.create(this.frameDownPins.sum(), ScoreType.NORMAL);
+        }
+
+        int nextBowlCount = this.frameDownPins.getScoreType().getBonusBowlCount();
+        List<Integer> nextDownPins = getNextDownPins(nextBowlCount);
+        if (nextDownPins.size() < nextBowlCount) {
+            return Score.create(0, ScoreType.READY);
+        }
+
+        int score = this.frameDownPins.sum() + nextDownPins.stream().reduce(0, Integer::sum);
+        return Score.create(score, ScoreType.NORMAL);
+    }
+
+    private boolean isLast() {
+        return this.frameNumber.getValue() == FINAL_FRAME_NUMBER;
+    }
+
+    private boolean isFinished() {
+        return !this.frameDownPins.hasTurn();
+    }
+
+    private List<Integer> getNextDownPins(int count) {
+        if (this.nextFrame == null) {
```

Comment:
```suggestion
        if (Objects,isNull(this.nextFrame)) {
```
Objects 클래스를 사용한다면 가독성을 높일수 있습니다 !